projects/angular2-jsonapi/src/services/json-api-datastore.service.ts

```diff
@@ -331,7 +331,7 @@ export class JsonApiDatastore {
     const models: T[] = [];
 
     body.data.forEach((data: any) => {
-      const model: T = this.deserializeModel(modelType, data);
+      const model: T = this.deserializeModel(this.datastoreConfig.models[data.type], data);
```

Comment:
could we still default to `modelType` though? So that it doesn't fail in the rare case there is no `type` field?

Reply:
yes, looks like it is a good idea

maybe it needs some documentation also, do not remember

anyway we ended up ejecting the library from our projects due to the lack of responsiveness to our MRs

so I do not feel like continuing with this one. I gonna close the MRs

thank you for your comment